lib/handlers/check_channel_connection_handler.dart

```diff
@@ -0,0 +1,82 @@
+import 'dart:async';
+
+import 'package:breez/bloc/account/account_bloc.dart';
+import 'package:breez/logger.dart';
+import 'package:breez/services/device.dart';
+import 'package:breez/services/injector.dart';
+import 'package:breez/theme_data.dart' as theme;
+import 'package:breez/widgets/flushbar.dart';
+import 'package:flushbar/flushbar.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+class CheckChannelConnection {
+  static final _instance = CheckChannelConnection._internal();
+
+  factory CheckChannelConnection() => _instance;
+
+  CheckChannelConnection._internal();
+
+  Timer _notReadyTimer;
+  Flushbar _flushbar;
+  StreamSubscription<bool> _subscription;
+
+  void startListen(
+    BuildContext context,
+    AccountBloc accountBloc,
+  ) {
+    ServiceInjector().device.eventStream.distinct().listen((event) {
+      switch (event) {
+        case NotificationType.RESUME:
+          _subscription = accountBloc.accountStream
+              .map((acc) => acc.readyForPayments)
+              .distinct()
+              .listen((ready) {
+            if (ready) {
+              _readyForPayments();
+              _notReadyTimer?.cancel();
+            } else {
+              _notReadyTimer = Timer(Duration(seconds: 30), () {
+                _notReadyForPayments(context);
+              });
+            }
+          });
+          break;
+        case NotificationType.PAUSE:
+          _subscription?.cancel();
```

Comment:
Good question, I did not add before because I did assume no, but I'm not sure the correct answer

Reply:
I think we need to remove, otherwise if from some reason the connection is restored in the background, upon resuming the flushbar will stay visible even though the channel is live.

Reply:
✅